### DISCUSSION TITLE
How to display cpu load of an attached domain

### DISCUSSION THREAD

**Discussion by Lenbok** [wezterm/wezterm#2265] [Q&A]
In tmux I have a plugin that displays the cpu load of the system that tmux is running on in the right status area. I would like similar functionality in wezterm. Is there a way I could say run a cron job on the remote system that sets a wezterm "domain-level" variable, that I could pick up in my "update-right-status" function?

**Comment by wez** [comment]
Nothing ideal, but for the general "how do I pass data from things running in a pane to the gui?" question, you may want to look at *user vars*

https://wezfurlong.org/wezterm/config/lua/pane/get_user_vars.html

The idea is that you do something in the precmd/prompt stuff in your shell on the remote host that might do something like:

```bash
printf "\033]1337;SetUserVar=%s=%s\007" UPTIME `uptime | base64`
```

Then in your right-status stuff you could access the current value of that var:

```lua
local uptime = pane:get_user_vars().UPTIME
```

The main limitation of this is that the uptime would only update whenever the shell ran the printf line, which would likely be each time the prompt was updated and wouldn't be during a long-running command.
